File: server/aoReporting/Views/DailyViewingsChartClass.vb

Option Strict On
Option Explicit On

Imports Contensive.Addons.Reporting.Controllers
Imports Contensive.BaseClasses

Namespace Views
    Public Class DailyViewingsChartClass
        Inherits AddonBaseClass
        '
        '=====================================================================================
        Public Overrides Function Execute(ByVal CP As CPBaseClass) As Object
            Dim result As String = ""
            Dim sw As New Stopwatch : sw.Start()
            Try
                Using ac As New applicationController(CP, False)
                    Dim Width As String = ac.cp.Doc.GetText("Width")
                    Dim Height As String = ac.cp.Doc.GetText("Height")
                    Dim durationHours As Integer = 24
                    Dim Rate As String = ac.cp.Doc.GetText("Rate")
                    If Rate.ToLower = "hourly" Then
                        durationHours = 1
                    End If
                    Dim DurationDays As Integer = ac.cp.Doc.GetInteger("Duration", 365)
                    Dim DivName As String = ac.cp.Doc.GetText("TargetDiv")
                    If DivName = "" Then
                        DivName = "PageViewChart"
                    End If
                    Dim DateEnd As Date = Now.Date

                    Dim DateStart As Date = DateEnd.AddDays(-DurationDays).Date
                    Dim dblDateStart As Double = DateStart.ToOADate()
                    Dim dblDateEnd As Double = DateEnd.ToOADate()
                    Dim criteria As String = "(TimeDuration=" & durationHours & ") AND (DateNumber>=" & dblDateStart & ") AND (DateNumber<" & dblDateEnd & ")"


                    Dim visitSummaryList As List(Of Models.visitSummaryModel) = Models.visitSummaryModel.createList(ac.cp, criteria, "TimeNumber desc")
                    If (visitSummaryList.Count = 0) Then
                        result = "<span class=""ccError"">There is currently no data collected to display this chart. Please check back later.</span>"
                    Else
                        result = Models.ChartViewModel.getChart(ac, visitSummaryList, DivName, False, Width, Height, (durationHours = 1))
                    End If
                End Using
            Catch ex As Exception
                CP.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace

File: server/aoReportingCSharp/AddonsLegacy/DailyViewingsChartClass.cs
using Contensive.BaseClasses;
using Contensive.Reporting.Models;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using DbBaseModel = Contensive.Models.Db.DbBaseModel;

namespace Contensive.Reporting {
    public class DailyViewingsChartClass : AddonBaseClass {
        // 
        // =====================================================================================
        public override object Execute(CPBaseClass CP) {
            string result = "";
            Stopwatch sw = new Stopwatch(); sw.Start();
            try {
                using (ApplicationModel ac = new ApplicationModel(CP, false)) {
                    string Width = ac.cp.Doc.GetText("Width");
                    string Height = ac.cp.Doc.GetText("Height");
                    int durationHours = 24;
                    string Rate = ac.cp.Doc.GetText("Rate");
                    if (Rate.ToLower() == "hourly")
                        durationHours = 1;
                    int DurationDays = ac.cp.Doc.GetInteger("Duration", 365);
                    string DivName = ac.cp.Doc.GetText("TargetDiv");
                    if (DivName == "")
                        DivName = "PageViewChart";
                    DateTime DateEnd = DateTime.Now.Date;

                    DateTime DateStart = DateEnd.AddDays(-DurationDays).Date;
                    double dblDateStart = DateStart.ToOADate();
                    double dblDateEnd = DateEnd.ToOADate();
                    string criteria = "(TimeDuration=" + durationHours + ") AND (DateNumber>=" + dblDateStart + ") AND (DateNumber<" + dblDateEnd + ")";


                    List<Models.VisitSummaryModel> visitSummaryList =  DbBaseModel.createList<VisitSummaryModel>(ac.cp, criteria, "TimeNumber desc");
                    if ((visitSummaryList.Count == 0))
                        result = "<span class=\"ccError\">There is currently no data collected to display this chart. Please check back later.</span>";
                    else
                        result = Models.ChartViewModel.getChart(ac, visitSummaryList, DivName, false, Width, Height, (durationHours == 1));
                }
            } catch (Exception ex) {
                CP.Site.ErrorReport(ex);
            }
            return result;
        }
    }
}
